featurebyte/service/task_manager.py

```diff
@@ -36,11 +37,30 @@ def __init__(
         persistent: Persistent,
         celery: Celery,
         catalog_id: Optional[ObjectId],
+        redis: Redis[Any],
     ) -> None:
         self.user = user
         self.persistent = persistent
         self.celery = celery
         self.catalog_id = catalog_id
+        self.redis = redis
+
+    @property
+    def periodic_task_service(self) -> PeriodicTaskService:
+        """
+        Get PeriodicTaskService instance
+
+        Returns
+        -------
+        PeriodicTaskService
+        """
+        return PeriodicTaskService(
+            user=self.user,
+            persistent=self.persistent,
+            catalog_id=self.catalog_id,
+            block_modification_handler=BlockModificationHandler(),
+            redis=self.redis,
+        )
```

Comment:
Curious why don't we add this to `__init__`?

Reply:
Currently periodic task service is catalog specific while task manager can be instantiated without a catalog. 
It may be possible to refactor this later to simplify the code here